src/ethereum/Wallet.js

```diff
@@ -0,0 +1,51 @@
+// Interface
+export class Wallet {
```

Comment:
I'm not sure `Wallet` is the best name for this. It's really hard to name as both the `web3` and `ledger` alternatives use the `web3` object, which itself is a bag of _things_ ranging from sending transactions to eth utilities.

Maybe `Client` is better?